File: frontend/src/api/openvpn/request/get_subnets.ts
import { api } from 'boot/axios';
import API from 'api/openvpn/api';
import Subnet from 'api/openvpn/data/subnet';

export interface IGetSubnetsResponse extends Response {
  subnets: Subnet[];
}

export default function getSubnets() {
  return api.get<IGetSubnetsResponse>(API.SUBNETS);
}

File: backend/openvpn/request/get_subnets.py
from dataclasses import dataclass, field

from api.base_request import BaseRequest
from api.base_response import BaseResponse
from openvpn.data.subnet import Subnet


@dataclass
class GetSubnetsRequest(BaseRequest):
    pass


@dataclass
class GetSubnetsResponse(BaseResponse):
    subnets: list[Subnet] = field(default_factory=list)
